Source/init.cpp

```diff
@@ -156,12 +156,12 @@ void __fastcall init_create_window(int nCmdShow)
 	wcex.style = CS_HREDRAW | CS_VREDRAW;
 	wcex.lpfnWndProc = WindowProc;
 	wcex.hInstance = ghInst;
-	wcex.hIcon = LoadIcon(ghInst, MAKEINTRESOURCE(IDI_ICON1));
-	wcex.hCursor = LoadCursor(0, IDC_ARROW);
+	wcex.hIcon = LoadIcon(ghInst, (LPCSTR)MAKEINTRESOURCE(IDI_ICON1));

```

Comment:
MAKEINTRESOURCE should be a macro, the current implementation is a bit of a hack where we instead have a function that maps error message and title indexes. So instead of casting things here we should correct DialogBoxParam and MAKEINTRESOURCE in https://github.com/diasurgical/devilutionX/blob/master/SourceX/DiabloUI/diabloui.cpp#L69